### DISCUSSION TITLE
create2

### DISCUSSION THREAD

**Discussion by sakulstra** [foundry-rs/foundry#3587] [Q&A] [answered]
Hello, trying to use `create2` with foundry i found foundry will use `0x4e59b44847b379578588920ca78fbf26c0b4956c` per default https://github.com/foundry-rs/foundry/blob/9d64f1fbc47d20f30c91e71c4560fd5b11d65a21/evm/src/executor/inspector/cheatcodes/util.rs#L26

I'm wondering how "safe" it is to use this deployer in terms of
- why is this address no verified anywhere?
- who deploys this?, will it be deployed on new chains? Currently it seems to deployed on all chains relevant for me, but e.g. metis, zksync etc are not and wondering if it makes more sense to deploy a own factory (not so as easy as i understand it as with custom factories verification might not work) https://github.com/foundry-rs/foundry/issues/2638
- in https://github.com/foundry-rs/foundry/issues/2638 there seems to be some discussion about a changed interface which will obviously change the deployer address, will the `0x4e59b44847b379578588920ca78fbf26c0b4956c` will still be maintained after that?

**Comment by mds1** [comment, marked as answer]
1. It's written in yul, I don't know if etherscan/others supports yul verification. You can find the source here: https://github.com/Arachnid/deterministic-deployment-proxy
2. That same link contains a tx that you can use to deploy to any network (modulo networks with different gas metering). But that tx will likely not deploy the contract to the same address on zkSync or other zkEVMs, since my understanding is that they'll likely use a different hash function than keccak (used for address calculations) since keccak is not zk-friendly and has a large proving overhead (though a quick search shows maybe zkSync does use keccak)
3. Re maintenance of existing deployer, per above, anyone can maintain it by deploying on new networks. Or if you mean maintained as in "still a supported option in foundry", that's TBD (since that issue is still TBD) and I'd suggest leaving a comment on #2638 about that

**Reply by sakulstra** [reply]
Don't know enough about this to have an opinion - quite literally the first time i'm using `create2` :sweat_smile:

Would be good imo to have a section in the docs essentially containing what you answered here.
Thx for quick response :pray:
